src/Components/WebAssembly/WebAssembly.Authentication/src/Services/BaseAddressAuthorizationMessageHandler.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+
+namespace Microsoft.AspNetCore.Components.WebAssembly.Authentication
+{
+    /// <summary>
+    /// A <see cref="DelegatingHandler"/> that will attach authorization tokens using the default scopes to
+    /// outgoing requests where the application base address is a base of the target request url.
+    /// </summary>
+    public class BaseAddressAuthorizationMessageHandler : AuthorizationMessageHandler
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="BaseAddressAuthorizationMessageHandler"/>.
+        /// </summary>
+        /// <param name="provider">The <see cref="IAccessTokenProvider"/> to use for requesting tokens.</param>
+        /// <param name="navigationManager">The <see cref="NavigationManager"/> used to compute the base address.</param>
+        public BaseAddressAuthorizationMessageHandler(IAccessTokenProvider provider, NavigationManager navigationManager)
+            : base(provider, navigationManager)
+        {
+            ConfigureHandler(new[] { navigationManager.BaseUri });
```

Comment:
How do you configure scopes or returnUrl if you're using `BaseAddressAuthorizationMessageHandler`? Seems like you can't call `ConfigureHandler` again from user code because it will throw saying it's already configured.

Is there any use case for configuring scopes/returnUrl with this?

Reply:
Not that I can think of, given that it uses the default set of scopes, and if you really want that, you can simply configure an AuthorizationMessageHandler yourself. The property here is for convenience.

Reply:
OK, sounds fine!